### ISSUE TITLE
Solution soundness issue on QF_BV formula

### ISSUE THREAD

**Issue by rainoftime** [stp/stp#383]
Hi, for the following formula,
stp 101aee2 yields sat
but both cvc4 and z3 return unsat
~~~~
(set-logic QF_BV)
(declare-const x (_ BitVec 8))
(declare-const y (_ BitVec 8))
(declare-const z (_ BitVec 8))
(assert (or (not (= (bvadd x y z) (bvadd (bvadd x y) z)))
            (not (= (bvmul x y z) (bvmul (bvmul x y) z)))
            (not (= (bvand x y z) (bvand (bvand x y) z)))
            (not (= (bvor x y z) (bvor (bvor x y) z)))
            (not (= (bvxor x y z) (bvxor (bvxor x y) z)))
            (not (= (bvxnor x y z) (bvxnor (bvxnor x y) z)))))
(check-sat)
~~~~

**Comment by TrevorHansen** [comment]
Thanks for letting us know.

It looks like STP doesn't think that bvxnor is associative, this returns
sat:
```
(set-logic QF_BV)
(declare-const x (_ BitVec 8))
(declare-const y (_ BitVec 8))
(declare-const z (_ BitVec 8))
(assert (not (= (bvxnor x y z) (bvxnor (bvxnor x y) z))))
(check-sat)
```
with a model of :
```
(model
( define-fun |x| () (_ BitVec 8) #x00 )
( define-fun |y| () (_ BitVec 8) #x00 )
( define-fun |z| () (_ BitVec 8) #x00 )
)
```

**Comment by rainoftime** [comment]
A discussion here https://github.com/Boolector/boolector/issues/150#issuecomment-698722418

**Comment by aniemetz** [comment]
As mentioned in the Boolector issue referenced above, operator XNOR was mistakenly marked as :left-associative in SMT-LIB, this has been corrected recently. The behavior for n-ary XNORs is not defined in SMT-LIB now.

**Comment by aniemetz** [comment]
CVC4 previously still treated bvxnor as left-associative, fixed here: CVC4/CVC4#5138
